api/librdb-api.h

```diff
@@ -330,6 +330,19 @@ _LIBRDB_API void RDB_setLogLevel(RdbParser *p, RdbLogLevel l);
 _LIBRDB_API void RDB_setLogger(RdbParser *p, RdbLoggerCB f);
 _LIBRDB_API void RDB_log(RdbParser *p, RdbLogLevel lvl, const char *format, ...);
 
+/* Following function returns a hint for the total number of items in the current
+ * parsed key context - to assist with memory allocation or other optimizations.
+ * Return valid value inside the following callbacks:
```

Comment:
please note that the existence of the hint depends on the encoding, and is not always available.

Reply:
I specified in the comment explictly in which callbacks it will return a valid value. Please lmk if you think it is not so.

Reply:
my argument is that it's not about which states or callbacks will have that data, i think the user should expect to get `-1` anyway (even in these callbacks).
i'm not even sure we need to list in which callbacks it may or may not be available (what we have today may change tomorrow), it's a hint that may or may not exist..